source/game.c

```diff
@@ -3517,22 +3517,29 @@ static void game_round_end_dismiss_round_end_panel()
 
 static void print_price_under_sprite_object(SpriteObject* sprite_object, int price)
 {
+    int width = 0;
+    int height = 0;
+
+    sprite_object_get_size(sprite_object, &width, &height);
+
     int x = fx2int(sprite_object->tx) + TILE_SIZE - (get_digits_even(price) - 1) * TILE_SIZE;
-    // TODO: Should probably extract the sprite size
-    int y = fx2int(sprite_object->ty) + CARD_SPRITE_SIZE + TILE_SIZE;
+    int y = fx2int(sprite_object->ty) + height + TILE_SIZE;
     tte_printf("#{P:%d,%d; cx:0x%X000}$%d", x, y, TTE_YELLOW_PB, price);
 }
 
-static void erase_price_under_sprite_object(SpriteObject* sprite_object)
+static void erase_price_under_sprite_object(SpriteObject* sprite_object, int price)
 {
+    int width = 0;
+    int height = 0;
+
+    sprite_object_get_size(sprite_object, &width, &height);
+
     Rect price_rect;
     price_rect.left = fx2int(sprite_object->tx);
-    price_rect.top = fx2int(sprite_object->ty) + CARD_SPRITE_SIZE + TILE_SIZE;
-    // TODO: Should probably extract the size from the sprite
-
-    price_rect.right = price_rect.left + TILE_SIZE * 3;
-    // Taking 2 tiles down so the highlighted case is also covered
-    price_rect.bottom = price_rect.top + 2 * TILE_SIZE;
+    // todo: fix the + 4 offset, don't know why its needed
```

Comment:
Please don't add nebulous magic fixes. The value 4 needs some context to it's purpose in either a variable/macro name or a comment.